src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -269,16 +270,17 @@ public IList<JsonPropertyInfo> Properties
         // Avoids having to perform an expensive cast to JsonTypeInfo<T> to check if there is a Serialize method.
         internal bool HasSerializeHandler { get; private protected set; }
 
-        // Configure would normally have thrown why initializing properties for source gen but type had SerializeHandler
+        // Exception used to throw on deserialization. In some scenarios i.e.:
+        // configure would have thrown while initializing properties for source gen but type had SerializeHandler
         // so it is allowed to be used for fast-path serialization but it will throw if used for metadata-based serialization
-        internal bool MetadataSerializationNotSupported { get; private protected set; }
+        internal Exception? DeserializationException { get; private protected set; }
```

Comment:
this is a bit unrelated, I changed this because I initially was planning to throw an exception when `required` was used and ended up changing it. I think it makes sense to do that. I didn't capture context here because it didn't feel like original exception stack would be much useful here